### PR TITLE
Fix issue with onRow

### PR DESCRIPTION
**Pull request by rehandalal**
Turns out I did it all wrong!

r?

### CODE REVIEW

File: src/components/tables/DataList.js [hunk 1]
```diff
@@ -60,7 +60,11 @@ export default class DataList extends React.PureComponent {
   }
 
   render() {
-    const { columnRenderers, columns, dataSource, onRow } = this.props;
+    const { columnRenderers, columns, dataSource, onRowClick } = this.props;
+
+    const onRow = (record, index) => ({
```
Comment (magopian) [inline src/components/tables/DataList.js]: Is there an `index`? According to https://ant.design/changelog there's only `record`

**Comment by rehandalal** [comment]
> Serves me right for not testing it myself

Me too! I just assumed it was a drop-in change.

File: src/components/tables/DataList.js (hunk 1 above)
Comment (rehandalal) [reply src/components/tables/DataList.js]: Yep, there is an index. They didn't do a great job documenting this.

**Comment by rehandalal** [comment]
bors r=@magopian

**Pull request merged**